src/app/rundown/components/segment/segment.component.ts

```diff
@@ -1,25 +1,103 @@
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core'
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnDestroy,
+  Output, SimpleChange,
+  SimpleChanges
+} from '@angular/core'
 import { Segment } from '../../../core/models/segment'
+import { Part } from '../../../core/models/part'
+import {PieceLayerService} from "../../../shared/services/piece-layer.service";
+import { PieceLayer } from '../../../shared/enums/piece-layer'
 
 @Component({
   selector: 'sofie-segment',
   templateUrl: './segment.component.html',
   styleUrls: ['./segment.component.scss']
 })
-export class SegmentComponent {
-
+export class SegmentComponent implements OnChanges, OnDestroy {
+  // TODO: Remove when we have change detection for reference values.
   @Input()
-  public isRundownActive: boolean
+  public isOnAir: boolean
 
   @Input()
   public segment: Segment
 
   @Output()
-  setNextEvent: EventEmitter<{segmentId: string, partId: string}> = new EventEmitter()
+  public setNextEvent: EventEmitter<{segmentId: string, partId: string}> = new EventEmitter()
```

Comment:
This comment also applies for `setNextEvent` on Part.

All this do is propagate up to the RundownComponent so it can do a call to `RundownService`, where it doesn't need to know the response.

Why not just call the RundownService directly from the Part component?
We have a RundownStateService to handle the state, so we can allow our "dumb" component to make this call for us.
Saves a lot of boilerplate code ;)

Reply:
If we remove the boilerplate, we just need to pass down the `rundownId` instead to the PartComponent. Is that desirable?

Reply:
`Segment` aleady have the `rundownId` on it, so you would only need the input for rundownId on the Part componenet, so it's not that bad.